apis/config/v1alpha1/common_types.go

```diff
@@ -188,6 +189,7 @@ type ACL struct {
 }
 
 func (a *ACL) Model() (models.ACL, error) {
+	sort.Strings(a.Values)
 	values := strings.Join(a.Values, " ")
```

Comment:
```suggestion
        var values string
        if len(a.Values) > defaults.MaxLineArgs-3 {
		values = fmt.Sprintf("-f %s", a.FilePath())
	} else {
	  values = strings.Join(a.Values, " ")
	} 
```

Reply:
It is also necessary to sort the values when len(a.Values) > defaults.MaxLineArgs-3 otherwise, the hash generated from the sorted values does not match the hash generated in the ACLValuesFiles #(https://github.com/six-group/haproxy-operator/blob/f0d238f0342ea244f1c7cea035b7d0886e281f46/controllers/instance/config.go#L413)

Reply:
ah got it 👍